services/app-api/src/handlers/bulk_download.ts

```diff
@@ -84,23 +80,37 @@ export const main: APIGatewayProxyHandler = async (event) => {
     // here we go through all the keys and open a stream to the object.
     // also, the filename in s3 is a uuid, and we store the original
     // filename in the content-disposition header. set original filename too.
-    const s3DownloadStreams: S3DownloadStreamDetails[] = await Promise.all(
-        bulkDlRequest.keys.map(async (key: string) => {
-            const params = { Bucket: bulkDlRequest.bucket, Key: key }
-            const headCommand = new HeadObjectCommand(params)
-            const metadata = await s3.send(headCommand)
-            const getCommand = new GetObjectCommand(params)
-            const filename = parseContentDisposition(
-                metadata.ContentDisposition ?? key
-            )
-            const s3Item = await s3.send(getCommand)
-            return {
-                stream: s3Item.Body as Readable,
-                key: key,
-                filename,
-            }
-        })
-    )
+    let s3DownloadStreams: S3DownloadStreamDetails[]
+    try {
+        s3DownloadStreams = await Promise.all(
+            bulkDlRequest.keys.map(async (key: string) => {
+                const params = { Bucket: bulkDlRequest.bucket, Key: key }
+                const headCommand = new HeadObjectCommand(params)
+                const metadata = await s3.send(headCommand)
+                const getCommand = new GetObjectCommand(params)
+                const filename = parseContentDisposition(
+                    metadata.ContentDisposition ?? key
+                )
+                const s3Item = await s3.send(getCommand)
+                return {
+                    stream: s3Item.Body as Readable,
+                    key: key,
+                    filename,
+                }
+            })
+        )
+    } catch (e) {
```

Comment:
:( I wonder if this is a change we should really have everywhere. Idk why I didn't notice we weren't try catching our async actions in handlers that should probably be our standard.

Also eslint should warn on this too? Gonna check that out